### DISCUSSION TITLE
I am getting NameError: name 'quit' is not defined. What to do ?

### DISCUSSION THREAD

**Discussion by mvpalheta** [aahnik/wappdriver#45] [Q&A] [answered]
I get that error when try run de code below

```python
from wappdriver import WhatsApp
with WhatsApp() as bot:
    bot.send('Eu_mesmo',  # name of recipient
             'hi send by a bot')  # message
```

**Comment by aahnik** [comment]
Hi @mvpalheta , thank you for asking.

- First of all check your python version by running `python --version`.
- I think you are running an old version of `python`. Please update python to latest 3.9 for best results.
-  `quit()` is a python exit command, which is built into python.
-  Please make sure that you dont have any variable or module named `quit`
- Please share the full traceback ( output in terminal ), for more help.

**Reply by mrwenjie** [reply]
I have encountered the same issue today. I think the problem with quit is caused by Ipython (Spyder, Jupyter Notebook...). If you run the code in PyCharm then it will work well.

**Comment by aahnik** [comment, marked as answer]
Thanks to @mrwenjie , for helping me figure out the actual problem.

Solution to the problem: **Don't use ipython. Create a python script, and then run that from terminal.**


For example create `script.py` and write the code in it. Then from terminal or command prompt, go to that directory and run `python script.py`

<details>
<summary> 👉 Click here to watch an ad </summary>

[![promoted](https://raw.githubusercontent.com/aahnik/aahnik/master/promoted/general.gif)](https://aahnik.github.io/aahnik/promoted/general.html)

</details>
